clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -707,7 +707,39 @@ getKernelInvocationKind(FunctionDecl *KernelCallerFunc) {
 
 static const CXXRecordDecl *getKernelObjectType(FunctionDecl *Caller) {
   assert(Caller->getNumParams() > 0 && "Insufficient kernel parameters");
-  return Caller->getParamDecl(0)->getType()->getAsCXXRecordDecl();
+
+  QualType KernelParamTy = (*Caller->param_begin())->getType();
```

Comment:
I don't know if you've done the re-merge yet, but this line changes a bit to Caller->getParamDecl(0)->getType().

Reply:
done.